tests/functional/codegen/features/iteration/test_for_range.py

```diff
@@ -427,6 +451,11 @@ def foo() -> DynArray[int256, 10]:
         res.append(i)
     return res
     """
-    c = get_contract(code)
-    with tx_failed():
-        c.foo()
+
+    if experimental_codegen:
+        with pytest.raises(StaticAssertionException):
+            get_contract(code)
+    else:
+        c = get_contract(code)
+        with tx_failed():
+            c.foo()
```

Comment:
we should change the code as above so the runtime check is appropriately checked